core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/deletevolumes/DeleteVolumesService.java

```diff
@@ -0,0 +1,118 @@
+                payload.getStackId().toString());
+    }
+
+    public void detachResources(List<CloudResource> cloudResourcesToBeDetached, String cloudPlatform, AuthenticatedContext ac) throws Exception {
+        LOGGER.debug("Detaching volumes {}", cloudResourcesToBeDetached);
+        CloudPlatformVariant cloudPlatformVariant = new CloudPlatformVariant(Platform.platform(cloudPlatform), Variant.variant(cloudPlatform));
+        CloudConnector cloudConnector = cloudPlatformConnectors.get(cloudPlatformVariant);
+        cloudConnector.volumeConnector().detachVolumes(ac, cloudResourcesToBeDetached);
+    }
+
+    public void deleteResources(List<CloudResource> cloudResourcesToBeDeleted, String cloudPlatform, AuthenticatedContext ac) throws Exception {
+        LOGGER.debug("Deleting volumes {}", cloudResourcesToBeDeleted);
+        CloudPlatformVariant cloudPlatformVariant = new CloudPlatformVariant(Platform.platform(cloudPlatform), Variant.variant(cloudPlatform));
+        CloudConnector cloudConnector = cloudPlatformConnectors.get(cloudPlatformVariant);
+        cloudConnector.volumeConnector().deleteVolumes(ac, cloudResourcesToBeDeleted);
+    }
+
+    public void updateCbdbResources(StackDto stackDto, DeleteVolumesRequest payload) throws Exception {
+        StackDeleteVolumesRequest stackDeleteVolumesRequest = payload.getStackDeleteVolumesRequest();
+        List<Resource> resourcesToBeDeleted = stackDto.getResources().stream().filter(resource -> null != resource.getInstanceGroup()
+                && resource.getInstanceGroup().equals(stackDeleteVolumesRequest.getGroup()) && resource.getResourceType()
+                .equals(ResourceType.AWS_VOLUMESET)).collect(Collectors.toList());
+        LOGGER.debug("Deleting resources from CBDB {}", resourcesToBeDeleted);
+        resourceService.deleteAll(resourcesToBeDeleted);
+        flowMessageService.fireEventAndLog(stackDto.getId(), Status.AVAILABLE.name(), CLUSTER_DELETEDVOLUMES, stackDeleteVolumesRequest.getGroup(),
+                stackDto.getId().toString());
+    }
+
+    public void stopClouderaManagerService(StackDto stackDto, Set<ServiceComponent> hostTemplateServiceComponents) throws Exception {
+        ClusterApi clusterApi = clusterApiConnectors.getConnector(stackDto);
+        for (ServiceComponent serviceComponent : hostTemplateServiceComponents) {
+            try {
+                LOGGER.debug("Stopping CM service {}, in stack {}", serviceComponent.getService(), stackDto.getId());
+                clusterApi.clusterModificationService().stopClouderaManagerService(serviceComponent.getService());
+                commonUtilService.pollClouderaManagerServices(clusterApi, serviceComponent.getService(), "STOPPED");
+            } catch (Exception e) {
+                LOGGER.error("Unable to stop CM services for service {}, in stack {}", serviceComponent.getService(), stackDto.getId());
+                throw new Exception(String.format("Unable to stop CM services for " +
```

Comment:
Can you please use a more related exception type here?

Reply:
done